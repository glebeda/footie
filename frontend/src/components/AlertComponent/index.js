import React from 'react';
import { Alert } from '@mui/material';

const AlertComponent = ({ message, severity, open, onClose }) => {
  if (!open) return null;

  return (
    <Alert severity={severity} onClose={onClose}>
      {message}
    </Alert>
  );
};

export default AlertComponent;
