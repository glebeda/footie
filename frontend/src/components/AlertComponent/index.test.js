import React from 'react';
import { render, screen } from '@testing-library/react';
import AlertComponent from './index';

describe('AlertComponent', () => {
  it('renders the message when open', () => {
    render(<AlertComponent message="Test Alert" severity="error" open={true} onClose={() => {}} />);
    expect(screen.getByText('Test Alert')).toBeInTheDocument();
  });

  it('does not render when not open', () => {
    const { container } = render(<AlertComponent message="Test Alert" severity="error" open={false} onClose={() => {}} />);
    expect(container).toBeEmptyDOMElement();
  });

  it('matches the snapshot when open', () => {
    const { asFragment } = render(<AlertComponent message="Test Alert" severity="error" open={true} onClose={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot when not open', () => {
    const { asFragment } = render(<AlertComponent message="Test Alert" severity="error" open={false} onClose={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
});