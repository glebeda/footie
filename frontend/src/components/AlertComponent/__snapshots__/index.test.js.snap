// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertComponent matches the snapshot when not open 1`] = `<DocumentFragment />`;

exports[`AlertComponent matches the snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-x7o5da-MuiPaper-root-MuiAlert-root"
    role="alert"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="ErrorOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      Test Alert
    </div>
    <div
      class="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="Close"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        title="Close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
